sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1084,12 +1086,29 @@ def _retry_region_zones(self,
             global_user_state.ClusterStatus.UP
         ]
 
-        self._clear_blocklist()
         for region, zones in self._yield_region_zones(to_provision,
                                                       cluster_name,
                                                       cluster_exists):
-            if self._in_blocklist(to_provision.cloud, region, zones):
+            # Filter out zones that are blocked, if any.
+            # This optimize the provision loop by skipping zones that are
+            # indicated to be unavailable from previous provision attempts.
+            # It can happen for the provisioning on GCP, as the
+            # yield_region_zones will return zones from a region one by one,
+            # but the optimizer that does the filtering will not be involved
+            # until the next region.
+            filtered_zones = copy.deepcopy(zones)
+            for zone in zones:
+                for blocked_resources in self._blocked_resources:
+                    if to_provision.copy(region=region.name,
+                                         zone=zone.name).should_be_blocked_by(
+                                             blocked_resources):
+                        filtered_zones.remove(zone)
+                        break
```

Comment:
We don't want to break twice, since we are filtering out zones that should be blocked, and the break here will skip the current zone and go to the next zone.